homeassistant/helpers/typing.py

```diff
@@ -35,23 +33,25 @@ class UndefinedType(Enum):
 UNDEFINED = UndefinedType._singleton  # noqa: SLF001
 
 
+def _deprecated_typing_helper(attr: str) -> DeprecatedAlias:
+    """Help to make a DeprecatedAlias."""
+    # pylint: disable-next=import-outside-toplevel
+    import homeassistant.core
+
+    return DeprecatedAlias(
+        getattr(homeassistant.core, attr), f"homeassistant.core.{attr}", "2025.5"
+    )
+
+
 # The following types should not used and
 # are not present in the core code base.
 # They are kept in order not to break custom integrations
 # that may rely on them.
 # Deprecated as of 2024.5 use types from homeassistant.core instead.
-_DEPRECATED_ContextType = DeprecatedAlias(
-    homeassistant.core.Context, "homeassistant.core.Context", "2025.5"
-)
-_DEPRECATED_EventType = DeprecatedAlias(
-    homeassistant.core.Event, "homeassistant.core.Event", "2025.5"
-)
-_DEPRECATED_HomeAssistantType = DeprecatedAlias(
-    homeassistant.core.HomeAssistant, "homeassistant.core.HomeAssistant", "2025.5"
-)
-_DEPRECATED_ServiceCallType = DeprecatedAlias(
-    homeassistant.core.ServiceCall, "homeassistant.core.ServiceCall", "2025.5"
-)
+_DEPRECATED_ContextType = _deprecated_typing_helper("Context")
```

Comment:
The function will be evaluated on import so this won't work it seems.

Reply:
Ugh, you're right. I'll change it so the evaluation is deferred.

Reply:
Fixed!